### ISSUE TITLE
HttpWebRequest not using system proxy settings

### ISSUE THREAD

**Issue by davidsh** [dotnet/corefx#31122]
While investigating other HttpClient/HttpWebRequest proxy-related bugs, I discovered that
HttpWebRequest was not honoring system proxy settings as defined on Windows with IE
settings or on Linux using environment variables.

Repro:
```c#
HttpWebRequest webRequest = WebRequest.CreateHttp("http://corefx-net.cloudapp.net/echo.ashx");
HttpWebResponse webResponse = webRequest.GetResponse() as HttpWebResponse;

using (Stream httpStream = webResponse.GetResponseStream())
{
    using (StreamReader httpReader = new StreamReader(httpStream))
    {
        string response = httpReader.ReadToEnd();
        Console.WriteLine(response);
     }
}
```

On Linux (but it repro's on Windows as well), set system proxy settings using 'http_proxy' environment variable.  Assume a proxy is running on that server.

```
export http_proxy=http://172.17.0.3:8888
```

When running against HttpClient, the request goes thru a proxy. You can see the 'Via' request header that was sent as sent back from the echo server.  But using HttpWebRequest, there is no 'Via' request header since the request did not go thru the proxy.
```
root@546974bdd7f0:/home/root/httptest# dotnet run
Sending request with HttpClient
200 OK
{"Method":"GET","Url":"/echo.ashx","Headers": {
    "Connection": "close",
    "Via": "1.1 tinyproxy (tinyproxy/1.8.4)",
    "Host": "corefx-net.cloudapp.net"
  },
  "Cookies": {},
  "BodyContent": "",
  "BodyLength": 0,
  "SecureConnection": false,
  "ClientCertificatePresent": false,
  "ClientCertificate": null
}
Sending request with HttpWebRequest
{"Method":"GET","Url":"/echo.ashx","Headers": {
    "Connection": "Keep-Alive",
    "Host": "corefx-net.cloudapp.net"
  },
  "Cookies": {},
  "BodyContent": "",
  "BodyLength": 0,
  "SecureConnection": false,
  "ClientCertificatePresent": false,
  "ClientCertificate": null
}
```

**Comment by davidsh** [comment]
Fixed in master with #31100

**Comment by danmosemsft** [comment]
This is approved but release/2.1 branch is not open for 2.1.4 yet so we can't merge yet.

**Comment by davidsh** [comment]
### Shiproom template

#### Description
HttpWebRequest is not honoring system proxy settings as defined on Windows with IE settings or on Linux using environment variables. So, sending requests with HttpWebRequest will ignore the proxy completely.

The problem is due to how HttpClient and HttpWebRequest differ in how they represent the default behavior of using system proxy settings with its various properties. On .NET Core, HttpWebRequest is a layer on top of HttpClient. Sending a request with HttpWebRequest will create a new HttpClient/HttpClientHandler object. However, the bug is that the proxy settings are not correctly translated from HttpWebRequest settings to HttpClient settings.

Although this bug is in HttpWebRequest, the bug did not manifest in .NET Core 2.0 on Linux because CurlHandler will always query the environment variables regardless of the .NET layer proxy settings (a different bug). So, the behavior on .NET Core 2.0 on Linux was that HttpWebRequest would respect system proxy settings. But with .NET Core 2.1 (using SocketsHttpHandler), this HttpWebRequest bug was now exposed.

#### Customer Impact
Without this fix, HTTP requests using HttpWebRequest will ignore system proxy settings. For some enterprises, that means that HTTP requests will not work at all if they are normally required to use the proxy.

#### Regression?
Regression from 2.0.

The behavior that the customer will see is a REGRESSION from .NET Core 2.0 to .NET Core 2.1 in using HttpWebRequest APIs. Overall, we have had a LOT of customers reporting problems in .NET Core 2.1 with HTTP related APIs and using system/environment proxy settings (NuGet restore, etc). Given the number of other proxy related bugs we’ve already fixed (and still have more to fix), we should consider fixing this for servicing as well.

#### Risk
The fix is low risk and the impact is high in terms of solving a key problem.

**Comment by gronlund013** [comment]
Since @davidsh closed my issue

Would like to mention that until this fix is present in a release, 2.1 is unusable for me and my teams.
We work in corporate environments daily and we deploy to corporate environments where proxy's and vpn's are almost always guaranteed..

I hope you fix this soon, I wish you good luck in this since we're not long from our next release that is based heavily on 2.1

Good luck boys and girls!

**Comment by davidsh** [comment]
This fix was approved for the future release/2.1.4 servicing release.  So, when that servicing update goes out, this bug will be fixed.

**Comment by joshfree** [comment]
@davidsh can you please merge the 2.1 PR and then resolve this issue? The checkin needs to happen before Tuesday to make 2.1.4. Thanks

**Comment by davidsh** [comment]
Fixed merged to release/2.1.x branch with #31123

**Comment by joshfree** [comment]
Thanks!
